### ISSUE TITLE
Feature Request: PsN Default Number of Threads

### ISSUE THREAD

**Issue by billdenney** [osmosisfoundation/pmx-docker#5]
PsN by default has 5 threads for NONMEM execution.  While it's possible to change this on the fly, it's preferable to me to have it default to the number of licenses I've purchased.  For that, I'd propose the following change to the Dockerfile:

```
# in the preamble
ARG NUM_THREADS=4
# After running setup.pl
  && sed s/threads=5/threads=$NUM_THREADS/ /usr/local/share/perl/*/PsN_4*/psn.conf > psn.conf.tmp \
  && mv psn.conf.tmp /usr/local/share/perl/*/PsN_4*/psn.conf \
```

**Comment by BretFisher** [comment]
Isn't this something we could set as a ENV with a default value and let someone overwrite at runtime, rather than worrying about build time? or does it affect builds/testing?

**Comment by dpastoor** [comment]
no should not affect builds/testing - an env variable with a default of 4 would be most reasonable, as that would be relevant for a single license pack.

**Comment by billdenney** [comment]
I set it as an ARG because users can input their number of licenses and then it becomes the default for all executions in their installation.  Is there a significant downside to an ARG with a default argument as opposed to an ENV?

**Comment by billdenney** [comment]
The effect on build is related to the config file setting.  Because it affects many commands within PsN, I think that this is the simplest way to do set the default number of threads (though I'm happy to be proven wrong).

**Comment by BretFisher** [comment]
Anything that is "environment specific" such as running on one machine vs. another should be a ENV that's changeable at runtime. An ARG is only used at build time, which is fine if the value never needs to change during runtime. Sounds like we might need a `docker-entrypoint.sh`, [which is the standard for how to change config files during container startup](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#entrypoint). We can have docker-compose.yml have the ENV in it, and document that anyone can change that value before startup, then `docker-entrypoint.sh` runs on startup and writes changes to config files.

**Comment by BretFisher** [comment]
OK so I've updated some things in master. Latest commits fix this issue I believe.

- I've added some getting started info to [README.md](https://github.com/osmosisfoundation/pmx-docker/blob/master/README.md#how-to-use) that may help understand this process.
- The assumption is the standard user is using `docker-compose` commands for this and using our pre-built images, but once someone knows what they are doing they can use the alternate compose file and an override file to tweak things.
- `NUM_THREADS` is now an ENV that writes to /root/psn.conf at container runtime. Just update the `docker-compose.yml` or create a `docker-compose.override.yml` to modify. No need to rebuild images.
